### PR TITLE
Limit obs repeat based on another values

### PR DESCRIPTION
**Pull request by lucyjemutai**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

File: src/components/encounter/ohri-encounter-form.component.tsx [hunk 1]
```diff
@@ -452,6 +465,8 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
     if (field.questionOptions.rendering == 'toggle') {
       value = value ? ConceptTrue : ConceptFalse;
     }
+    if (field.id == 'repeatLimitField') {
+    }
```
Comment (samuelmale) [inline src/components/encounter/ohri-encounter-form.component.tsx:469]: Do you need this?

File: src/components/encounter/ohri-encounter-form.component.tsx [hunk 2]
```diff
@@ -492,6 +507,34 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
           );
         }
 
+        if (
+          dependant.questionOptions.rendering === 'repeating' &&
+          !isEmpty(dependant.questionOptions.repeatOptions?.limitExpression)
+        ) {
+          //('limit expression', dependant.questionOptions.repeatOptions?.limitExpression);
```
Comment (samuelmale) [inline src/components/encounter/ohri-encounter-form.component.tsx:514]: Get rid of the above comment

File: src/components/repeat/ohri-repeat.component.tsx [hunk 3]
```diff
@@ -54,10 +54,10 @@ export const updateFieldIdInExpression = (expression: string, index: number, que
   return expression;
 };
 
-export const showAddButton = (repeatOptions: { limit?: string }, counter: number) => {
-  return isEmpty(repeatOptions?.limit) || Number(repeatOptions.limit) === 0 || typeof repeatOptions.limit === 'string'
-    ? true
-    : counter < Number(repeatOptions.limit);
+export const showAddButton = (limit: string | number, counter: number) => {
+  const repeatLimit = Number(limit);
+  //console.log('repeatLimit', repeatLimit);
```
Comment (samuelmale) [inline src/components/repeat/ohri-repeat.component.tsx:59]: Same here

**Review by samuelmale — CHANGES_REQUESTED** [review]
Left a couple of comments

**Comment by lucyjemutai** [comment]
yeah

File: src/components/repeat/ohri-repeat.component.tsx [hunk 4]
```diff
@@ -208,10 +208,10 @@ export const OHRIRepeat: React.FC<OHRIFormFieldProps> = ({ question, onChange })
...
     nodes.push(
       <div>
-        {showAddButton(question.questionOptions.repeatOptions, counter) && (
-          <Button
-            renderIcon={() => <Add size={16} />}
-            kind="ghost"
+        {showAddButton(question.questionOptions.repeatOptions?.limit, counter) && (
+                  <Button
+                    renderIcon={() => <Add size={16} />}
+                    kind="ghost"
```
[the remark this answers is not in the document]
Comment (samuelmale) [reply src/components/repeat/ohri-repeat.component.tsx:213]: @lucyjemutai what happens when you run:

```yarn run lint ```

**Pull request closed without merging**